Realm/ObjectStore/object_store.cpp

```diff
@@ -355,6 +362,26 @@ void ObjectStore::create_tables(Group *group, Schema &target_schema, bool update
             set_primary_key_for_object(group, target_object_schema->name, "");
         }
     }
+    return to_delete;
+}
+
+void ObjectStore::remove_properties(Group *group, Schema &target_schema, std::vector<Property> to_delete) {
+    for (auto& target_object_schema : target_schema) {
+        TableRef table = table_for_object_type(group, target_object_schema.name);
+        ObjectSchema current_schema(group, target_object_schema.name);
+        size_t deleted = 0;
+        for (auto& current_prop : current_schema.properties) {
+            current_prop.table_column -= deleted;
+            for (auto& single_to_delete : to_delete) {
+                if (current_prop.name == single_to_delete.name &&
+                    current_prop.object_type == single_to_delete.object_type) {
```

Comment:
`object_type` is the type of the linked object for object/array properties, not the type of the containing object.